src/renderer/views/Channel/Channel.vue

```diff
@@ -1048,6 +1063,34 @@ const videoContinuationData = shallowRef(null)
 const showVideoSortBy = ref(true)
 const videoSortBy = ref('newest')
 
+const filteredVideos = computed(() => {
+  const filteredArray = hideWatchedToggle.value ? filterWatchedArray(latestVideos.value) : latestVideos.value
+  if (hideWatchedToggle.value && filteredArray.length === 0) {
+    if (process.env.SUPPORTS_LOCAL_API && apiUsed === 'local') {
+      getChannelVideosLocalMore()
+    } else {
+      channelInvidiousVideos()
+    }
+  }
+  return filteredArray
```

Comment:
Computeds should not have side-effects, please remove the fetch more stuff out if the computed.

From the Vue 3 docs but it also applies to Vue 2: https://vuejs.org/guide/essentials/computed.html#getters-should-be-side-effect-free